### ISSUE TITLE
Slow performance while zooming

### ISSUE THREAD

**Issue by bilby91** [mapbox/DEPRECATED-mapbox-ios-sdk#531]
I'm having a lot of performance issues while zooming a map that has 4 lines and 2 polygons. The lines and polygons don't have more than 6 points. I'm experiencing this problem with a zoom level of more than 8, with zoom levels smaller than this it works smoothly. 

I'v tried reducing the calls to `correctPositionOfAllAnnotationsIncludingInvisibles:animated:` by calling it only on `scrollViewDidEndZooming:withView:atScale` but the problem persists, although it feels better.

I'm running iOS 8, have test it on an iPad Mini.

**Comment by incanus** [comment]
@bilby91 Can you provide a performance trace of what you're seeing? Or maybe provide your data?

**Comment by bilby91** [comment]
@incanus I have been trying to find the root of the problem and it seems to happen only when I configure the `lineDashLengths`, if I use the default one the performance is not affected. 

I'm using just two lines, only one has a lineDashLengths of @[ 2.5, 2.5 ]

```
Line 1
<+34.95799637,-128.84765625> +/- 0.00m (speed -1.00 mps / course -1.00) @ 11/5/14, 5:23:31 PM Uruguay Summer Time,
<+40.91351318,-102.78808594> +/- 0.00m (speed -1.00 mps / course -1.00) @ 11/5/14, 5:23:31 PM Uruguay Summer Time,
<+29.11377525,-85.16601562>

Line 2
<+32.10118866,-122.51953125> +/- 0.00m (speed -1.00 mps / course -1.00) @ 11/5/14, 5:23:53 PM Uruguay Summer Time,
<+40.27952576,-113.29101562> +/- 0.00m (speed -1.00 mps / course -1.00) @ 11/5/14, 5:23:53 PM Uruguay Summer Time,
<+34.27083588,-106.08398438> +/- 0.00m (speed -1.00 mps / course -1.00) @ 11/5/14, 5:23:53 PM Uruguay Summer Time,
<+34.08906174,-100.06347656> +/- 0.00m (speed -1.00 mps / course -1.00) @ 11/5/14, 5:23:53 PM Uruguay Summer Time
)
```

**Comment by incanus** [comment]
Ah, that certainly narrows it down. I'll try to take a look at this, but it sounds like it's likely just a Core Graphics rendering bottleneck and can't be helped.

**Comment by incanus** [comment]
So @bilby91, I don't see this on smaller shapes, but yes, on higher zooms with such a large shape (I tried your _Line 2_), I do see a performance impact with line dashes on. No problem with them off. In fact, I can even reliably crash my retina iPad by turning `scaleLineDash = YES` on for the shape. 

There's nothing special that we're doing here, just passing the dash parameters on to `-[CAShapeLayer lineDashPattern]`, so I think it's just a factor of the shape size and the fact that in reality it is huge at high zoom levels, despite not being entirely on screen. 

I experimented a bit with some settings like `shouldRasterize`, `drawsAsynchronously`, and having `RMMapOverlayView` (the parent view/layer) `maskToBounds`, but no effect. 

There's probably not a particularly easy way to enhance the performance here, if at all. I would recommend an approach like prerendering raster tiles for the shape in question or awaiting Mapbox GL, which will use hardware GPU rendering for things like this and be much more performant.
